orderbook/src/orderbook.rs

```diff
@@ -85,6 +89,52 @@ impl OrderBook {
         orders.retain(|_, order| has_future_valid_to(now_in_epoch_seconds, &order.order_creation));
     }
 
+    async fn remove_settled_orders(&self, settlement_contract: &GPv2Settlement) {
+        let uids_to_be_removed = self.get_uids_of_settled_orders(settlement_contract).await;
+        info!(
+            "removing following settled orders from orderbook: {:?}",
+            uids_to_be_removed
+        );
+        let mut orders = self.orders.write().await;
+        orders.retain(|uid, _| uids_to_be_removed.contains(uid));
+    }
+
+    async fn get_uids_of_settled_orders(
+        &self,
+        settlement_contract: &GPv2Settlement,
+    ) -> Vec<OrderUid> {
+        let orders = self.orders.read().await;
+        let uid_futures = orders.iter().map(|(uid, _)| async move {
+            (
+                *uid,
+                self.has_not_yet_been_settled(*uid, settlement_contract)
```

Comment:
ubernit: this could just return `Option<OrderUid>`, Some signalling this order has been settled.

Reply:
I finally understood your suggestion. Yeah. that is cool. And then the filter_map is also easier. I will try to do it.